src/EFCore.PG/Query/Expressions/Internal/PostgresAllExpression.cs

```diff
@@ -50,7 +50,7 @@ public class PostgresAllExpression : SqlExpression, IEquatable<PostgresAllExpres
             [CanBeNull] RelationalTypeMapping typeMapping)
             : base(typeof(bool), typeMapping)
         {
-            if (!array.Type.IsArrayOrGenericList())
+            if (!(array.Type.IsArrayOrGenericList() || array is SqlConstantExpression constant && constant.Value is null))
```

Comment:
```suggestion
            if (!(array.Type.IsArrayOrGenericList() || array is SqlConstantExpression { Value is null })
```

Reply:
OK OK, I need to start getting used to these :)